batch-models/src/main/scala/org/ekstep/analytics/exhaust/OnDemandDruidExhaustJob.scala

```diff
@@ -174,8 +174,17 @@ object OnDemandDruidExhaustJob extends BaseReportsJob with Serializable with IJo
             // TO:DO
             // Fetch report config from dataset_metadata table
             val datasetConf = getDataSetDetails(requestType)
-            val reportConf = if(datasetConf.druid_query.nonEmpty) JSONUtils.deserialize[Map[String,AnyRef]](datasetConf.druid_query.get) else JSONUtils.deserialize[Map[String,AnyRef]](AppConf.getConfig("druid_query." + requestType))
+            var reportConf = if(datasetConf.druid_query.nonEmpty) JSONUtils.deserialize[Map[String,AnyRef]](datasetConf.druid_query.get) else JSONUtils.deserialize[Map[String,AnyRef]](AppConf.getConfig("druid_query." + requestType))
             val sortDfColNames = reportConf.get("sort").asInstanceOf[Option[List[String]]]
+            // Date Range with dynamic Start Date and End Date
+            if (reportConf.contains("programDashboardDateRange").&(reportConf.getOrElse("programDashboardDateRange",None) == true)) {
+              val startDateVal = requestParamsBody.getOrElse("start_date",reportConf.getOrElse("defaultStartDate",None))
+              val endDateVal = requestParamsBody.getOrElse("end_date",reportConf.getOrElse("defaultEndDate",None))
```

Comment:
defaultEndDate - has to be defaulted to current date right? Why is it taken from reportConf.

Reply:
Can't it be implemented this way - if start_date & end_date are available from request, just replace it or else it can be left as it is(similar to earlier implementation) ? Why is extra boolean check **programDashboardDateRange** is needed here?

Reply:
this is how we have implemented, may be programDashboardDateRange boolean check i will remove.

The logic we will be following is :- if start_date & end_date is present in the request will be replace it, if end_date is not present in the request will replace it with the current_date and if start_date is not present in the request will replace it with  default start date from the report config. 
Hope this works

Thanks

Reply:
@sowmya-dixit its done. Please check and merge the PR